compose/config/fields_schema_v2.json

```diff
@@ -41,6 +41,13 @@
             "^.+$": {"type": ["string", "number"]}
           },
           "additionalProperties": false
+        },
+        "external": {
+          "type": ["boolean", "object"],
+          "properties": {
+            "name": {"type": "string"}
+          },
+          "additionalProperties": false
```

Comment:
We had a version of it with build vs image that you can look at (it's gone now, but it was in 1.5.2).  I'm not sure if there is any other way, other than using  oneOf: `requires`, `not: requires`